src/file.js

```diff
@@ -24,22 +24,8 @@ Sk.builtin.file = function (name, mode, buffering) {
     } else if (this.name === "/dev/stderr") {
         this.fileno = 2;
     } else {
-        if (Sk.inBrowser) {  // todo:  Maybe provide a replaceable function for non-import files
```

Comment:
Ah, I wish.  @brianpmarks writes all our good code.  :)

Things like file handling get pretty idiosyncratic, so I doubt our code is useful to anyone else. Either way, here are some snippets for how we do file writing:

https://gist.github.com/eah13/a6a805afbfca15031893409d936b5df1

Reading is similar.  You'll see a lot in there that's specific how we do file tabs but I think Runestone could totally do something like this for your ActiveCode widgets.